### DISCUSSION TITLE
Jquery in Next.js Application

### DISCUSSION THREAD

**Discussion by Sammy-White** [vercel/next.js#35940] [Help]
Is it right to use jQuery in next.js application. Since next.js allows server side rendering. I tried using Jquery in one of my next.js project but ran into issues.

**Comment by icyJoseph** [comment]
How did you try to use `jQuery`?

Often I have 3rd party providers that themselves load `jQuery` and things work just fine, of course, we don't use `jQuery` on the server ourselves. The result of server side rendering is the so called dry HTML, which is a string. To try and manipulate the DOM, on the server with `jQuery` shouldn't work I think. It constitutes unguarded DOM access, akin to reading from window directly.

Inside `useEffect`, event handlers and most promises, you should be fine though.

That being said, Next.js builds on React, which is what you should be using to construct your DOM. Mixing paradigms, and ownership of the DOM can only result in mysterious bugs, and hard to maintain code.
